pkg/operator/ceph/controller/spec.go

```diff
@@ -394,9 +394,37 @@ func ContainerEnvVarReference(envVarName string) string {
 }
 
 // DaemonEnvVars returns the container environment variables used by all Ceph daemons.
-func DaemonEnvVars(image string) []v1.EnvVar {
+func DaemonEnvVars(cephClusterSpec *cephv1.ClusterSpec) []v1.EnvVar {
+	var rookMsgr2EnvValue string
```

Comment:
The value doesn't really matter as long as it's unique for the settings, so how about if we do it with a single string format like this?
```go
fmt.Sprintf(
  "msgr2_%t_encryption_%t_compression_%t", 
  cephClusterSpec.Network.Connections.RequireMsgr2,
  //encryption setting,
  //compression setting)
```
IIRC you'll also need to check if cephClusterSpec.Network.Connections is nil. If that is nil, we can skip adding the env var.

Reply:
Let's just keep the string without any special char in between like `_` otherwise it will make code check longer unneccesarliy if the string value doesn't matter

Reply:
Can you show a code snippet? I thought it would just be a single `fmt.Sprintf()` as in my example, except checking for nil settings before calling `fmt.Sprintf()`.

Reply:
for example, let's say I remove the check `cephClusterSpec.Network.Connections.Compression != nil `then from the unit test you see it will through nil pointer which makes sense also right given these are struct settings apart from `requireMsgr2` which is just a bool

```
--- FAIL: TestDaemonEnvVars (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2619358]

goroutine 37 [running]:
testing.tRunner.func1.2({0x288d0c0, 0x3ea5220})
	/usr/local/go/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1548 +0x397
panic({0x288d0c0?, 0x3ea5220?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/rook/rook/pkg/operator/ceph/controller.ApplyNetworkEnv(0xc0000a3bc0)
	/home/srai/go/src/github.com/rook/pkg/operator/ceph/controller/spec.go:416 +0x78
github.com/rook/rook/pkg/operator/ceph/controller.TestDaemonEnvVars(0xc0005031e0?)
	/home/srai/go/src/github.com/rook/pkg/operator/ceph/controller/spec_test.go:487 +0x1a5
testing.tRunner(0xc0005031e0, 0x2ca3328)
	/usr/local/go/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 1
	/usr/local/go/src/testing/testing.go:1648 +0x3ad
FAIL	github.com/rook/rook/pkg/operator/ceph/controller	0.015s
FAIL
```

Reply:
What about like this? So it's a single format command once all the bools are determined.
```go
if cephClusterSpec.Network.Connections != nil {
  msgr2Required := false
  encryptionEnabled := false
  compressionEnabled := false
  if cephClusterSpec.Network.Connections.RequireMsgr2 {
    msgr2Required = true
  } 
  if cephClusterSpec.Network.Connections.Encryption != nil && cephClusterSpec.Network.Connections.Encryption.Enabled{
    encryptionEnabled = true
  }
  if cephClusterSpec.Network.Connections.Compression != nil && cephClusterSpec.Network.Connections.Compression.Enabled{
    compressionEnabled = true
  }
  envVarValue := fmt.Sprintf("msgr2_%t_encryption_%t_compression_%t", msgr2Required, encryptionRequired, compressionRequired)
  // create the env var
}
```